Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -974,6 +974,16 @@ bool WebPageProxy::shouldUseBackForwardCache() const
     return m_preferences->usesBackForwardCache() && backForwardCache().capacity() > 0;
 }
 
+void WebPageProxy::generateSyntheticCommandP(CompletionHandler<void(bool)>&& callbackFunction)
+{
+    if (!hasRunningProcess()) {
+        callbackFunction(false);
+        return;
+    }
+    
+    sendWithAsyncReply(Messages::WebPage::GenerateSyntheticCommandP(), WTFMove(callbackFunction));
```

Comment:
This seems to hard-code the command bindings of the web browser.

Is command-P the shortcut for printing worldwide in all the various translated versions of Safari?

I think it’s really sad that the websites hard code a keyboard equivalent, but it seems like this could entirely break printing for some Safari localizations.

Reply:
> Is command-P the shortcut for printing worldwide in all the various translated versions of Safari?

Yes, it is (Command+P isn't localized in Safari)/

> Here’s a testing idea: Customize the menu key shortcut for Print, changing it to something like Command-T, using the Keyboard Settings, Keyboard Shortcut, App Shortcuts, and see what happens.

Yup, I already tested this. Specifically, I tested:
* if it works in the DVORAK keyboard layout
* if it works for when the user changes the print command key binding (to `Command+Option+L` for example)
* for a non-Latin system language and keyboard (specifically, Hebrew)

and all of these worked. 

Note that I also have a corresponding Safari PR which is linked in the Radar. The Safari change will trigger the normal print flow if the command+P event was not handled